### PR TITLE
Add replaced and overridden states support for TACACS server resource module

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
SUMMARY
This is to add "replaced" and "overridden" states support for sonic_tacacs_server resource module.

GitHub Issues
"N/A".

ISSUE TYPE
- Feature Pull Request

COMPONENT NAME
sonic_tacacs_server

OUTPUT
- regression test result: 
[regression-2023-04-13-10-48-02.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11236497/regression-2023-04-13-10-48-02.html.pdf)
- regression test log: 
[regression_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11236498/regression_test.log)
- unit test log: 
[unit_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11236501/unit_test.log)
- unit test script: 
[tacacs.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11236503/tacacs.txt)

Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

How Has This Been Tested?
Tested on real SONIC switchs.

### CODE REVIEW

File: plugins/modules/sonic_tacacs_server.py [hunk 1]
```diff
@@ -249,8 +250,110 @@
...
+#timeout           : 10
+#auth-type         : pap
+#key configured    : Yes
+#--------------------------------------------------------------------------------------
+#HOST                 AUTH-TYPE    KEY-CONFIG PORT       PRIORITY   TIMEOUT    VRF
+#--------------------------------------------------------------------------------------
+#1.2.3.4              pap          No         49         1          5          default
+#11.12.13.14          chap         Yes        49         10         5          default
+#
+- name: Overridden tacacs configurations
```
Comment (stalabi1) [inline plugins/modules/sonic_tacacs_server.py:321]:
```suggestion
- name: Override tacacs configurations
```

File: tests/regression/roles/sonic_tacacs_server/defaults/main.yml [hunk 2]
```diff
@@ -85,7 +85,67 @@ tests:
...
+            timeout: 38
+            priority: 4
+          - name: 18.21.22.23
+            auth_type: chap
+            port: 20
+            timeout: 19
+            priority: 8
+
+  - name: test_case_08
+    description: overridden parameter of tacacs servers
```
Comment (stalabi1) [inline tests/regression/roles/sonic_tacacs_server/defaults/main.yml:129]:
```suggestion
    description: Override parameter of tacacs servers
```
Please capitalize the test case descriptions.

Comment (mingjunzhang2019) [reply tests/regression/roles/sonic_tacacs_server/defaults/main.yml]: Changed. Thanks!

File: plugins/modules/sonic_tacacs_server.py [hunk 3]
```diff
@@ -123,7 +124,7 @@
       - In case of merged, the input mode configuration will be merged with the existing tacacs server configuration on the device.
       - In case of deleted the existing tacacs server mode configuration will be removed from the device.
     default: merged
```
Comment (stalabi1) [inline plugins/modules/sonic_tacacs_server.py:126]: Please add descriptions for replaced and overridden.

Comment (mingjunzhang2019) [reply plugins/modules/sonic_tacacs_server.py:128]: Added. Thanks!

**Review by stalabi1 — APPROVED** [review]

**Review by kerry-meyer — APPROVED** [review]

**Pull request merged**